Maker-Agents/OSMAgent/src/addresses.fetcher.spec.ts

```diff
@@ -5,67 +5,95 @@ import { formatBytes32String, Interface } from "ethers/lib/utils";
+          inputs: [key],
+          outputs: [address.toLowerCase()],
+        });
       });
-    });
+    }
   };
 
   it("should store only PIP_ prefixed addresses", async () => {
     initialize();
-    await fetcher.getOsmAddresses(1);
+    for (let block = 1; block <= CONTRACTS.length; block++) {
+      await fetcher.getOsmAddresses(block);
 
-    expect(fetcher.osmContracts).toStrictEqual(
-      new Map<string, string>([
-        [formatBytes32String("PIP_ONE"), createAddress("0xa1")],
-        [formatBytes32String("PIP_TWO"), createAddress("0xa2")],
-        [formatBytes32String("PIP_THREE"), createAddress("0xa3")],
-      ])
-    );
+      expect(fetcher.osmContracts).toStrictEqual(
+        new Map<string, string>([
+          [formatBytes32String("PIP_ONE"), CONTRACTS[block - 1].get(formatBytes32String("PIP_ONE")) as string],
+          [formatBytes32String("PIP_TWO"), CONTRACTS[block - 1].get(formatBytes32String("PIP_TWO")) as string],
+          [formatBytes32String("PIP_THREE"), CONTRACTS[block - 1].get(formatBytes32String("PIP_THREE")) as string],
+        ])
+      );
+    }
   });
 
   it("should update the addresses correctly", async () => {
     initialize();
-    await fetcher.getOsmAddresses(1);
-    fetcher.updateAddresses("UpdateAddress", [formatBytes32String("PIP_FOUR"), createAddress("0xa7")]);
-    expect(fetcher.osmContracts).toStrictEqual(
-      new Map<string, string>([
-        [formatBytes32String("PIP_ONE"), createAddress("0xa1")],
-        [formatBytes32String("PIP_TWO"), createAddress("0xa2")],
-        [formatBytes32String("PIP_THREE"), createAddress("0xa3")],
-        [formatBytes32String("PIP_FOUR"), createAddress("0xa7")],
-      ])
-    );
 
-    fetcher.updateAddresses("RemoveAddress", [formatBytes32String("PIP_ONE")]);
-    expect(fetcher.osmContracts).toStrictEqual(
-      new Map<string, string>([
-        [formatBytes32String("PIP_TWO"), createAddress("0xa2")],
-        [formatBytes32String("PIP_THREE"), createAddress("0xa3")],
-        [formatBytes32String("PIP_FOUR"), createAddress("0xa7")],
-      ])
-    );
+    for (let block = 1; block <= CONTRACTS.length; block++) {
+      await fetcher.getOsmAddresses(block);
+
+      fetcher.updateAddresses("UpdateAddress", [formatBytes32String("PIP_FOUR"), createAddress("0xa7")]);
+      expect(fetcher.osmContracts).toStrictEqual(
+        new Map<string, string>([
+          [formatBytes32String("PIP_ONE"), CONTRACTS[block - 1].get(formatBytes32String("PIP_ONE")) as string],
+          [formatBytes32String("PIP_TWO"), CONTRACTS[block - 1].get(formatBytes32String("PIP_TWO")) as string],
+          [formatBytes32String("PIP_THREE"), CONTRACTS[block - 1].get(formatBytes32String("PIP_THREE")) as string],
+          [formatBytes32String("PIP_FOUR"), createAddress("0xa7")],
+        ])
+      );
+
+      fetcher.updateAddresses("RemoveAddress", [formatBytes32String("PIP_ONE")]);
```

Comment:
This is passing `"RemoveAddress"` to `updateAddresses`, however, the `if/else` inside of `fetcher` checks to see whether the event is `"UpdateAddress"` or not. So, in theory, this argument can be anything as long as it is not `"UpdateAddress"`.

If the intention is to **only** detect `"UpdateAddress"` or `"RemoveAddress"`, shouldn't the `if/else` only check to see if it is passed either one of those. Not simply if it is `"UpdateAddress"` or not `"UpdateAddress"`?

Also, is there a possibility of an address being removed if it detects an event that is neither `"UpdateAddress"` nor `"RemoveAddress"`? If so, is that a desired quality?